api/catalog/api/views/media_views.py

```diff
@@ -52,7 +52,17 @@ def get_queryset(self):
     # Standard actions
 
     def list(self, request, *_, **__):
-        self.paginator.page_size = request.query_params.get("page_size")
+        raw_page_size = request.query_params.get("page_size")
+        if (
+            request.user
+            and request.user.is_anonymous
+            and raw_page_size is not None
+            and int(raw_page_size) > 20
+        ):
+            raise get_api_exception(
+                "Page size must be between 0 & 20 for unauthenticated requests.", 400
```

Comment:
```suggestion
                "Page size must be between 0 & 20 for unauthenticated requests.", 401
```

401 is Unauthorized, which I think is more accurate here than "bad request" :slightly_smiling_face: